### PR TITLE
refactor: integrate CSS HMR into theme editor

### PR DESCRIPTION
**Pull request by sissbruecker**
## Description

Integrates HMR into theme editor:
- Remove loading of the full theme editor CSS into preview stylesheet
- Instead do an optimistic update by adding CSS rules for single property updates to preview stylesheet
- Change CSS HMR to dispatch a DOM event after
  - Vite has updated style modules (Vite hot deploy mode)
  - Dev tools have updates styles (Dev bundle mode)
- Listen for that event in theme editor, and when it is triggered, remove optimistic style updates, and refresh property values shown in editor. This should ensure that the correct property values are shown in all cases (change property, clear property, change class name, undo, redo).
- Removes mechanism for preventing live reload from the dev tools

Covers both dev bundle mode and Vite hot deploy mode.

Based on the changes from https://github.com/vaadin/flow/pull/16455 and currently set to merge against it.

### CODE REVIEW

**Review by MarcinVaadin — DISMISSED** [review]
Works fine after last commit, LGTM.

File: flow-server/src/main/resources/plugins/application-theme-plugin/theme-generator.js [hunk 1]
```diff
@@ -245,6 +245,10 @@ if (import.meta.hot) {
       window.location.reload();
     }
   });
+
+  import.meta.hot.on('vite:afterUpdate', (update) => {
```
Comment (Artur-) [inline flow-server/src/main/resources/plugins/application-theme-plugin/theme-generator.js:249]: This should be inside `if (import.meta.hot)` above

**Review by Artur- — CHANGES_REQUESTED** [review]

File: flow-server/src/main/resources/plugins/application-theme-plugin/theme-generator.js (hunk 1 above)
Comment (Artur-) [reply flow-server/src/main/resources/plugins/application-theme-plugin/theme-generator.js:249]: Oh, it actually is inside that...

**Review by Artur- — DISMISSED** [review]

**Comment by sissbruecker** [comment]
Rebased the PR on top of https://github.com/vaadin/flow/pull/16455, and added dispatching of the theme change event to the dev tools as well.

**Comment by Artur-** [comment]
rebased

**Review by MarcinVaadin — APPROVED** [review]
LGTM, tested and work. Support for _themeEditorLoadPreview_ command can be removed from debug message handler.

**Pull request merged**